City-of-Helsinki/drupal-helfi-kasvatus-koulutus
# UHF-4528 Large amount of units listing, Daycare search

# Add a way to list large amounts of units on a listing, Daycare search [UHF-4528](https://helsinkisolutionoffice.atlassian.net/browse/UHF-4528) & [UHF-4532](https://helsinkisolutionoffice.atlassian.net/browse/UHF-4532) 
Make it possible for the content producers to tag and list large numbers of units at the same time and using this feature add daycare search for KASKO.

## What was done
* Added views bulk operations and views bulk edit modules.
* Added Unit type vocabulary for the categorisation.
* Added new field Unit type for TPR-units.
* Changed the TPR-unit listing so that you can do views bulk edit on it.
* Added minimal display for TPR units.
* Added a new paragraph called daycare and enabled it for landing page and basic page bottom content.

## How to install
* Checkout this branch on your KASKO-instance.
* Take the platform config and hdbt to go with this PR.
    * `composer require drupal/helfi_platform_config:dev-UHF-4528_large_amount_of_units_listing && composer require drupal/hdbt:dev-UHF-4528_large_amount_of_units_listing`
* `composer install`
* `make drush-cr drush-cim drush-cr drush-updb drush-cr`
* `make shell`
* `drush locale:check && drush locale:update && drush cr`

## How to test
* Go to `/admin/structure/taxonomy/manage/unit_type/overview` and add one term called _Daycare_. This is important because the daycare search relies on this exact term name to work.
* Go to `/admin/content/integrations/tpr-unit` listing and check that there is action called _Modify values_ on the bottom of the listing. Select some units and do the action _Modify values_ for those units. Pick the term you just created for the _Unit type_ field on the selected units and save.
* Go to check any of the units that you edited if there is the term set on the _Unit type_ field.
* Now go to landing page or basic page and add the daycare search. There should be title and description fields for the paragraph. Save the node and it should now list all the units that you attached the term to.
* The block should look similar as the designs [here](https://share.goabstract.com/7d4f3697-821d-447c-aed2-3192dcb98d8d) with minor exceptions such as the "Tulokset järjestettynä hakemasi osoitteen perusteella" text position on desktop and the size of it in mobile. These were discussed with Sauli while developing the feature.
* Also test the address search that is a feature developed in the [UHF-4457](https://helsinkisolutionoffice.atlassian.net/browse/UHF-4475) story.

## Other PRs
* https://github.com/City-of-Helsinki/drupal-helfi-platform-config/pull/279
* https://github.com/City-of-Helsinki/drupal-hdbt/pull/270
* https://github.com/City-of-Helsinki/drupal-module-helfi-tpr/pull/106
